src/main/java/frc/robot/vision/multivisionsources/MultiVisionSources.java

```diff
@@ -0,0 +1,69 @@
+package frc.robot.vision.multivisionsources;
+
+import frc.constants.VisionConstants;
+import frc.robot.vision.data.VisionData;
+import frc.robot.vision.sources.VisionSource;
+import org.littletonrobotics.junction.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+
+public class MultiVisionSources<ReturnType extends VisionData> {
+
+	private final String logPath;
+	private final List<VisionSource<ReturnType>> visionSources;
+
+	@SafeVarargs
+	public MultiVisionSources(String logPath, VisionSource<ReturnType>... visionSources) {
+		this(logPath, List.of(visionSources));
+	}
+
+	public MultiVisionSources(String logPath, List<VisionSource<ReturnType>> visionSources) {
+		this.logPath = logPath;
+		this.visionSources = visionSources;
+	}
+
+	public ArrayList<ReturnType> getUnfilteredVisionData() {
+		return createMappedCopyOfSources(visionSources, VisionSource::getVisionData);
+	}
+
+	public ArrayList<ReturnType> getFilteredVisionData() {
+		return createMappedCopyOfSources(visionSources, VisionSource::getFilteredVisionData);
+	}
+
+	public void periodic() {
+		log();
+	}
+
+	protected List<VisionSource<ReturnType>> getVisionSources() {
```

Comment:
why have protected getter instead of making the variable protected?

Reply:
1. nahh it is better to have a getter
2. I don't really see a use for this method anyway, it can be deleted